### ISSUE TITLE
[drift] Allow to specify a query to filter projects in the hierarchy

### ISSUE THREAD

**Issue by smaftoul** [abcxyz/guardian#319]
### TL;DR

Cloudasset api allows specifying query filters.
Would be nice to allow passing a filter to the project listing query for 2 reasons:

* I reached several time the quota of `ListProjectsRequest`
* In the org I manage, there is a folder where we give developers developement projects where they are owner. In these project, we don't care about the drift.

### Detailed design
I added:

```go
var query string
if assetType == "cloudresourcemanager.googleapis.com/Project" {
        query = "state:ACTIVE AND parentAssetType:Organization OR (parentAssetType:Folder AND NOT parentFullResourceName:a-folder-id)"
} else {
        query = "state:ACTIVE"
}
```
at https://github.com/abcxyz/guardian/blob/6c77ec23ce573df4a49818788a7aa1bc0c1d41a7/pkg/assetinventory/assetinventory.go#L244-L248

and updated the query in `SearchAllResources` which just did what I needed.

### Additional information

I am interested in implementing properly this feature ;)

**Comment by smaftoul** [comment]
@dcreey Thanks for the suggestion. I tried it before creating the issue, and it didn't work.
I retried it again since you commented, and I thought I may have made a mistake ...

The thing is, I still have the message:

```
{"time":"...","severity":"DEBUG","logging.googleapis.com/sourceLocation":{"function":"github.com/abcxyz/guardian/pkg/commands/drift.(*IAMDriftDetector).DetectDrift","file":".../guardian/pkg/commands/drift/drift.go","line":148},"message":"fetching iam for org, folders and projects","number_of_folders":XX,"number_of_projects":XXX}
```

And still fetches the projects with `ListProjectsRequest`, which gets above quota.

Thus the suggestion to add a filter at the cloud asset API level.

**Comment by gjonathanhong** [comment]
@smaftoul Are you able to share the exact error message that you are receiving? I'm looking for any potential information that can help identify which exact API (e.g. `resourcemanager` or `cloudasset`), request and quota is failing. Looking at the drift implementation, we shouldn't be calling `ListProjects` directly.

**Comment by gjonathanhong** [comment]
Should be fixed by #332.

Feel free to reopen if the problem still persists.
